### PR TITLE
Install bundles with custom deployers and edit featuresBoot

### PR DESCRIPTION
**Pull request by hbfernandes**
- Enable bundles with custom deployers to be installed. 
- Enable assembly property edits to change featuresBoot by moving the application of the properties to the end.

Will any of these changes cause issues?

### CODE REVIEW

File: profile/src/main/java/org/apache/karaf/profile/assembly/Builder.java [hunk 1]
```diff
@@ -617,6 +608,15 @@ public String getValue(String key) {
         // Installed stage
         //
         installStage(installedProfile, allBootFeatures);
+
```
Comment (jbonofre) [inline profile/src/main/java/org/apache/karaf/profile/assembly/Builder.java]: For the record, this is already fixed on `master` and `karaf-4.1.x` branches.

File: profile/src/main/java/org/apache/karaf/profile/assembly/Builder.java [hunk 2]
```diff
@@ -1023,6 +1023,11 @@ private Profile startupStage(Profile startupProfile) throws Exception {
         for (Map.Entry<Integer, Set<String>> entry : invertedStartupBundles.entrySet()) {
             String startLevel = Integer.toString(entry.getKey());
             for (String location : new TreeSet<>(entry.getValue())) {
+                // remove any custom deployer prefix
+                int mvnIndex = location.indexOf("mvn:");
```
Comment (jbonofre) [inline profile/src/main/java/org/apache/karaf/profile/assembly/Builder.java]: Same question as in the other PR: I don't understand why you remove the `mvn` prefix from the bundle location.

Comment (hbfernandes) [reply profile/src/main/java/org/apache/karaf/profile/assembly/Builder.java]: Like i said on the other PR, `mvn:` is not being removed.

File: profile/src/main/java/org/apache/karaf/profile/assembly/Builder.java (hunk 1 above)
Comment (hbfernandes) [reply profile/src/main/java/org/apache/karaf/profile/assembly/Builder.java]: Ok, thats good. Still needed this part in to make my scenario work though.

**Pull request merged**